modelstore/storage/local.py

```diff
@@ -43,37 +43,38 @@ class FileSystemStorage(BlobStorage):
         "optional": [],
     }
 
-    def __init__(self, root_path: Optional[str] = None):
+    def __init__(self, root_dir: Optional[str] = None):
         super().__init__([])
-        root_path = environment.get_value(root_path, "MODEL_STORE_ROOT")
-        if MODELSTORE_ROOT in root_path:
+
+        if MODELSTORE_ROOT_PREFIX in root_dir:
             warnings.warn(
-                f'Warning: "{MODELSTORE_ROOT}" is in the root path, and is a value'
+                f'Warning: "{MODELSTORE_ROOT_PREFIX}" is in the root path, and is a value'
                 + " that this library usually appends. Is this intended?"
             )
-        root_path = os.path.abspath(root_path)
+        root_path = os.path.abspath(root_dir)
         self.root_dir = root_path
         logger.debug("Root is: %s", self.root_dir)
 
     def validate(self) -> bool:
         """This validates that the directory exists
         and can be written to"""
         # pylint: disable=broad-except
-        try:
-            parent_dir = os.path.split(self.root_dir)[0]
-            # Check that directory exists
-            if not os.path.exists(parent_dir):
-                logger.error("Error: %s does not exist", parent_dir)
-                return False
+        parent_dir = os.path.split(self.root_dir)[0]
 
+        if not os.path.exists(parent_dir):
+            raise Exception("Error: Parent directory to root dir '%s' does not exist", parent_dir)
```

Comment:
This has changed the return type - from returning `False` to raising an `Exception`. What motivated this? Happy to consider it for the next release - curious for your thoughts

Reply:
Am also thinking about whether `os.path.exists()` is enough, as we probably also need to check `os.path.isdir()`

Reply:
Added in https://github.com/operatorai/modelstore/pull/116